gateway/default.conf

```diff
@@ -11,39 +11,111 @@ server {
     try_files $uri $uri/ /index.html;
   }
 
+  # Internal auth route only accessible by NGINX
+  location = /auth {
+    internal;
+    proxy_pass http://auth-service:8005;
+    proxy_pass_request_body off;
+    proxy_set_header Content-Length "";
+  }
+
+  # Public routes
+  location ~ ^/(signin|verify-jwt|signout)$ {
+    proxy_pass http://auth-service:8005;
+  }
+
+  # Public routes
+  location /api/user-service/signup {
+    proxy_pass http://user-service:8000/api/user/signup;
+  }
+
   location /api/user-service {
+    # No authentication required for preflight requests
+    if ($request_method = OPTIONS) {
+      add_header Access-Control-Allow-Origin 'http://localhost:3000';
```

Comment:
I think this should change to frontend URL instead, or you could make it dynamically change based on environment, dev or prod?